.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,59 @@
+name: Publish image to Github Packages
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: | # The first 4 lines are default; have to include in order to specify 5th (latest)
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=tag
+            type=ref,event=pr
+            type=raw,value=latest
```

Comment:
I think, but am not sure, that we can delete the `tags:` key because I only had this previously so that pushes to branches would get tagged with `latest` (probably bad practices as I'm realizing we should interpret `latest` as "latest stable". I believe that pushes to `main` get tagged with `latest` by default using this workflow (even with nothing specified under `tags`, though it's quite opaque via this configuration interface.

Reply:
In fact if we are going to enable manual workflow dispatch, we *should* delete this (at least the final entry) because we don't want test pushes to get tagged as `latest`.